### ISSUE TITLE
skip tables drush is failing

### ISSUE THREAD

**Issue by MuschPusch** [omega8cc/boa#641]
This is working on our dev instances but not at the omega stack:

```drush sql-dump --skip-tableslist=cache,cache_block,cache_bootstrap,cache_field,cache_filter,cache_form,cache_libraries,cache_menu,cache_metatag,cache_page,cache_path,cache_token,cache_update,cache_views,cache_views_data,session --result-file=/some-path/file.sql --verbose```

**Comment by omega8cc** [comment]
What you mean by "dev instances" and "omega stack" ?

**Comment by MuschPusch** [comment]
Sorry. It's working on a non omega8cc server with drush7 and a bash. It's not working on the hosted BOA instances provided by you.

**Comment by omega8cc** [comment]
if this is a difference between self hosted BOA and BOA hosted on our servers, you need to report this at https://omega8.cc/support and not here.

**Comment by omega8cc** [comment]
Unless you mean non-BOA server? Please try to provide more complete report, so we don't need to ask extra questions like this.

**Comment by MuschPusch** [comment]
oh dear... I have a communication problem (non-boa related) Let me try again:

- non-boa server (BOA not installed) with drush 7 and bash the drush command works fine
- on the hosted boa provided by you it fails. Most likely it will also fail on a self hosted boa instance but i can't test that

**Comment by omega8cc** [comment]
OK, but you can't compare bash account with lshell account on BOA, right? Limited shell may have its limitations.

Please provide more details, like the complete error displayed, path used etc and not just a statement "skip tables drush is failing".

Alternatively, please report this via https://omega8.cc/support if you think it is only specific to our hosted BOA and not to BOA in general.

**Comment by omega8cc** [comment]
OK, I just tried this on a demo server and got this: https://gist.github.com/omega8cc/e75d228d4f3b54fa72a2

**Comment by omega8cc** [comment]
But, this failed also when using o1.ftp with bash, so it is not related to the shell used.

**Comment by omega8cc** [comment]
I believe it is related to the fact that we delegate all cache tables to Redis and Drush is not aware of this, perhaps. There is also no point in excluding any cache tables in `sql-dump` because they are empty anyway. It is enough to exclude any other non-cache related tables, if you need this, like `--skip-tables-list=session`

**Comment by omega8cc** [comment]
This works just fine:

````
o1.ftp:~$ drush @site sql-dump --skip-tables-list=session --result-file=/data/disk/o1/static/stest.sql
Database dump saved to /data/disk/o1/static/stest.sql [success]
o1.ftp:~$
````

**Comment by MuschPusch** [comment]
You cheat :)

```drush @site sql-dump --skip-tables-list=session, actions --result-file=/data/disk/o1/static/stest.sql
Database dump saved to /data/disk/o1/static/stest.sql```

Will fail

**Comment by MuschPusch** [comment]
Some debug output:
```
drush sql-dump --skip-tables-list="sessions,actions"  --result-file=/home/o531xxxxx.ftp/tmp/xxxx-live.sql --verbose
Initialized Drupal 7.34 root directory at /data/disk/xxx/static/xxx/public                                                                                                             [notice]
Initialized Drupal site bla.com at sites/bla.com                                                                                                        [notice]
Executing: mysql --defaults-extra-file=/home/o5xxx.ftp/.tmp/drush_McjImd --database=xxxx --host=localhost --port=3306 --silent  < /home/xxxxx1.ftp/.tmp/drush_RMulDl
Calling system((mysqldump --defaults-extra-file=/home/o53xxxx.ftp/.tmp/drush_injS84  xxxx --host=localhost --port=3306 --no-autocommit --single-transaction --opt -Q --ignore-table=xxx.actions --ignore-table=wxxxx.sessions) > /home/o531xxxx.ftp/tmp/live.sql);
Database dump failed                                                                                                                                                                             [error]
Command dispatch complete

**Comment by omega8cc** [comment]
Interesting. It works with `--skip-tables-list=session` but fails with `--skip-tables-list=session,actions`. Still, it is not related to shell used. Could be a Drush bug, I think. We are using Drush 7 head in BOA, by the way.

Let's see if newer Drush still fails like this.

**Comment by omega8cc** [comment]
It doesn't work with vanilla Drush7 (head) and with just bash, not lshell, so I doubt it is a BOA specific problem. It would be great if someone could confirm this using latest Drush7 head on any non-BOA system.
